Orange-OpenSource/its-client
# 1.4.0: MobilePerceivedObject RSU computation + SPATEM/MAPEM

What's new ?
------------------

* Message no longer implements Mobile trait
* `as_mobile()` method added to Message to access Mobile when manipulating generic Message
* SPATEM struct
* MAPEM struct
* Extend PerceivedObject management to RSU originating CPM
* Bump version to 1.4.0

How to test
---------------
1. Build and test the lib and examples
    ```
    cargo clean; cargo build --all; cargo test --all
    ```
    **=> Build must success**  
    **=> All tests must pass**  

2. Start local mosquitto server
    ```
    systemctl start mosquitto
    ```

3. Start its-client
    ```
    cargo run --bin its-client -- --mqtt-client-id unique_id_111 --root-topic 5GCroCo/inQueue
    ```

4. Start the Paul Sabatier's single vehicle SUMO scenario
    ```
    ./start.sh -H localhost -P 1883 --mqtt-client-id sumo_ps_nbu -c ../sumo-paul-sabatier/sumo_data/static_vehicle.sumocfg
    ```
    **=> On the HMI you must see two vehicles instead of one (the second being a 3 seconds delayed copy of the first one)**
